scripts/boot-p2p.py

```diff
@@ -17,184 +17,180 @@
-    # signing_key = nacl.signing.SigningKey.generate() 
-    # verify_key = signing_key.verify_key
-    # encoded_private_key = signing_key.encode(encoder=nacl.encoding.Base16Encoder).lower()
-    # encoded_public_key = verify_key.encode(encoder=nacl.encoding.Base16Encoder).lower() 
-
-    # import ed25519
-    # signing_key, verifying_key = ed25519.create_keypair()
-    # encoded_private_key = signing_key.to_ascii(encoding="base16").decode('utf-8')
-    # encoded_public_key = verifying_key.to_ascii(encoding="base16").decode('utf-8')
-
-    # import ecdsa # secp256k1 suppport
-
-    # Using pre-generated validator key pairs by rnode. We do this because warning below  with python generated keys
-    # WARN  coop.rchain.casper.Validate$ - CASPER: Ignoring block 2cb8fcc56e... because block creator 3641880481... has 0 weight
-    f=open('scripts/demo-validator-private-public-key-pairs.txt')
-    lines=f.readlines()
-    line_number = random.randint(1,295)
-    encoded_private_key = lines[line_number].split()[0]
-    encoded_public_key = lines[line_number].split()[1]
-
-    # print(f"Populating bonds file {bonds_file}")
-    # bond_weight = random.randint(1,100)
-    # line = f"{encoded_public_key} {bond_weight}"
-    # print(line)
-    # try:
-    #     with open(bonds_file, 'a+') as f:
-    #         f.write(f"{line}\n")
-    # except IOError as e:
-    #     print(f"Failed to open or write to file {e}.") 
-
-    return encoded_private_key, encoded_public_key
-
-
-def populate_bonds_file():
-    print(f"Populating bonds file {bonds_file}")
-    try:
-        with open(bonds_file, 'a+') as f:
-            for line in bonds:
-                print(line)
-                f.write(f"{line}\n")
-    except IOError as e:
-        print(f"Failed to open or write to file {e}.") 
+    import ed25519
+    while True:
+        signing_key, verifying_key = ed25519.create_keypair()
+        encoded_private_key = signing_key.to_ascii(encoding="base16").decode('utf-8')
+        encoded_public_key = verifying_key.to_ascii(encoding="base16").decode('utf-8')
+        yield KeyPair(private_key=encoded_private_key, public_key=encoded_public_key)
+    # with open(keys_file) as f:
+    #     lines = f.readlines()
+    # random.shuffle(lines)
+    # for line in lines:
+    #     yield KeyPair(*line.split())
+
+
+key_pairs_gen = generate_validator_private_key()
+
+
+def modify_bonds_file():
+    # In order to produce a bonds.txt which contains the key pair of the bonded node which is going to start,
+    # the bonded nodes key pair should be put into the bonds file
+    # So I have to add the key pairs to the bonds file.
+    print('Modify bonds file and add bonds key pair to the bond file')
+    with open(args.bonds_file) as bond_f:
+        with open(temp_bonds_file, 'w') as f:
+            f.write(bond_f.read())
+            for i in range(args.sigs):
+                key_pair = next(key_pairs_gen)
```

Comment:
Does `key_pairs_gen` have to be a generator?  IIUC, it could be an ordinary function.

Reply:
Yes~! You are right.